stdlib/public/core/StringUTF8.swift

```diff
@@ -208,27 +208,46 @@ extension String {
     public struct Index : Comparable {
       internal typealias Buffer = _StringCore._UTF8Chunk
 
-      init(_ _core: _StringCore, _ _coreIndex: Int,
-           _ _buffer: Buffer) {
-        self._core = _core
+      init(_ _coreIndex: Int, _ _buffer: Buffer) {
```

Comment:
@gribozavr Will this still be hidden? `Buffer` look like it's an alias to an underscored type, but it's just `UInt64` IIRC.